### PR TITLE
Mihály Szigyártó

### PR DESCRIPTION
**Pull request by szigya**
### Short Description

Mihály Szigyártó - Solution for interview task, fixed some typos.
Sorry for the big commit, I used bitbucket at work and I could only merge one commit, but if you want I can split into multiple. It seems a bit disturbing on github.

### Checks

- [x] This PR follows the coding guidelines of the project
- [x] This PR fulfills the acceptance criteria defined on the issue
- [x] This PR has been locally tested
- [x] Changes in this PR are unit tested

### CODE REVIEW

File: src/main/java/ch/aaap/assignment/Application.java [hunk 1]
```diff
@@ -74,8 +95,18 @@ public long getAmountOfPoliticalCommunitiesInDistict(String districtNumber) {
    */
   public LocalDate getLastUpdateOfPoliticalCommunityByPostalCommunityName(
       String postalCommunityName) {
-    // TODO implementation
-    throw new RuntimeException("Not yet implemented");
+
+    model.getPostalCommunityByName(postalCommunityName);
+
+    var postalCommunity = model.getPostalCommunityByName(postalCommunityName);
```
Comment (robertblust) [inline src/main/java/ch/aaap/assignment/Application.java:101]: Looks like you take care about the evolution of java and use new available features 👍

File: src/main/java/ch/aaap/assignment/model/Model.java [hunk 2]
```diff
@@ -1,14 +1,75 @@
 package ch.aaap.assignment.model;
 
+import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
-public interface Model {
+public class Model {
```
Comment (robertblust) [inline src/main/java/ch/aaap/assignment/model/Model.java:7]: Makes sense to change interface to class cause we do not expose the model to to external consumers and we just have 1 implementation. As well nicely separation of Model and the Model creation with the transformer 👍

File: src/main/java/ch/aaap/assignment/transformer/CSVDataToModelTransformer.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import ch.aaap.assignment.raw.CSVPostalCommunity;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Transforms raw csv data ( described as a set of {@link CSVPoliticalCommunity} and set of {@link
+ * CSVPostalCommunity}) to a {@link Model}.
+ */
+public class CSVDataToModelTransformer {
```
Comment (robertblust) [inline src/main/java/ch/aaap/assignment/transformer/CSVDataToModelTransformer.java:23]: nice and clean. easy to read 😄

File: src/main/java/ch/aaap/assignment/Application.java [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
 import ch.aaap.assignment.raw.CSVPoliticalCommunity;
 import ch.aaap.assignment.raw.CSVPostalCommunity;
 import ch.aaap.assignment.raw.CSVUtil;
+import ch.aaap.assignment.transformer.CSVDataToModelTransformer;
 import java.time.LocalDate;
+import java.util.Collection;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public class Application {
```
Comment (robertblust) [inline src/main/java/ch/aaap/assignment/Application.java:17]: usage of java streams 👍 as we expect 😄

**Comment by szigya** [comment]
> 
> 
> @szigya well done 👍 clean code, good solution and I like the approach with the model and the transformer. As well the additional test cases for the transformer. This is a critical part and well tested.
> 
> The only small remark I have is about code formatting. I have some white space differences e.g.
> Screenshot 2020-05-18 at 08 19 03
> But this is whine at a high level 😄
> 
> You will be contacted by Simona till the middle of next week.
> 
> Cheers
> 
> @robertblust

All right, thank you very much.

**Pull request closed without merging**